### PR TITLE
Do not build cli tools for iphone targets

### PR DESCRIPTION
**Pull request by alloy**
## Summary

Currently (in `v0.7.0`) we build the CLI tools for both `iphone` and `macos` targets, which means that, when installing into the same location, cmake will skip installing the CLI tools from the second build. In our current situation, the second installation is the macOS build, which is really the only target from which we want CLI tools.

```
$ tar -zxvf hermes-engine-darwin-0.7.0.tgz
$ file package/destroot/bin/hermesc
package/destroot/bin/hermesc: Mach-O universal binary with 3 architectures: [arm_v7:Mach-O executable arm_v7] [arm_v7s:Mach-O executable arm_v7s] [arm64]
package/destroot/bin/hermesc (for architecture armv7):    Mach-O executable arm_v7
package/destroot/bin/hermesc (for architecture armv7s):    Mach-O executable arm_v7s
package/destroot/bin/hermesc (for architecture arm64):    Mach-O 64-bit executable arm64
```

Put differently, nobody runs `hermesc` on iOS, we always only want the macOS builds of these.

----

I opted to make building CLI tools a simple cmake option and then use that from the apple builds to control wether or not to include them.

## Test Plan

With the `hermes-engine-darwin-0.7.0.tgz` artefact of this `npm` CI job:

```
$ tar -zxvf hermes-engine-darwin-0.7.0.tgz
$ file package/destroot/bin/hermesc
package/destroot/bin/hermesc: Mach-O universal binary with 2 architectures: [x86_64:Mach-O 64-bit executable x86_64] [arm64]
package/destroot/bin/hermesc (for architecture x86_64):	Mach-O 64-bit executable x86_64
package/destroot/bin/hermesc (for architecture arm64):	Mach-O 64-bit executable arm64
```

### CODE REVIEW

**Comment by alloy** [comment]
cc @grabbou

File: CMakeLists.txt [hunk 1]
```diff
@@ -591,14 +594,17 @@ set(CMAKE_CXX_EXTENSIONS OFF)
...
 add_subdirectory(include)
 add_subdirectory(lib)
 add_subdirectory(public)
 add_subdirectory(external)
 add_subdirectory(API)
 add_subdirectory(android/intltest/java/com/facebook/hermes/test)
 
+if(HERMES_ENABLE_TOOLS)
+  add_subdirectory(tools)
+endif()
```
Comment (grabbou) [inline CMakeLists.txt:606]: Looks like there are few changes in this diff that are unrelated styling changes? (apart from this one ofc)

File: utils/build-apple-framework.sh [hunk 2]
```diff
@@ -27,6 +27,13 @@ fi
 
 # Utility function to configure an Apple framework
 function configure_apple_framework {
+  local build_cli_tools
+  if [[ $1 == iphone* ]]; then
+    build_cli_tools="false"
+  else
+    build_cli_tools="true"
+  fi
```
Comment (grabbou) [inline utils/build-apple-framework.sh:35]: Personally I think having the `if` check for `mac` and explicitly setting it to `true` (`false` otherwise) would be a bit more future proof when we add e.g. Apple Watch.

**Review by grabbou — APPROVED** [review]
LGTM. Good catch! Left a few comments, but I like this change.

File: CMakeLists.txt (hunk 1 above)
Comment (alloy) [reply CMakeLists.txt:606]: I’m unsure which changes you mean, can you elaborate?

File: utils/build-apple-framework.sh (hunk 2 above)
Comment (alloy) [reply utils/build-apple-framework.sh:35]: Good point 👍

**Comment by alloy** [comment]
@tmikov @Huxpro This is good to go 👍

File: CMakeLists.txt (hunk 1 above)
Comment (alloy) [reply CMakeLists.txt:606]: @grabbou I think you might be referring to all the whitespace changes below (?), which are because of the additional `if` check.

**Comment by grabbou** [comment]
Small conflict in a `.sh` file because of me :) Should easy to resolve tho.

File: .circleci/config.yml [hunk 3]
```diff
@@ -45,6 +45,7 @@ apple_defaults: &apple_defaults
   working_directory: ~/hermes
   environment:
     - TERM: dumb
+    - HERMES_WS_DIR: /tmp/hermes
     # Homebrew currently breaks while updating:
     # https://discuss.circleci.com/t/brew-install-fails-while-updating/32992
     - HOMEBREW_NO_AUTO_UPDATE: 1
```
Comment (rolsen949) [inline .circleci/config.yml:51]:
```suggestion
    - HOMEBREW_NO_AUTO_UPDATE: 1
```

File: .circleci/config.yml [hunk 4]
```diff
@@ -195,14 +196,7 @@ jobs:
...
   macos:
-    macos:
-      xcode: "12.0.0-beta"
-    environment:
-      - HERMES_WS_DIR: /tmp/hermes
-      - TERM: dumb
-      # Homebrew currently breaks while updating:
-      # https://discuss.circleci.com/t/brew-install-fails-while-updating/32992
-      - HOMEBREW_NO_AUTO_UPDATE: 1
+    <<: *apple_defaults
```
Comment (rolsen949) [inline .circleci/config.yml:199]: _

Comment (rolsen949) [inline .circleci/config.yml:199]:
```suggestion
    <<: *apple_defaults
```

File: .circleci/config.yml [hunk 5]
```diff
@@ -317,14 +311,7 @@ jobs:
...
     # CheckedMalloc.Death fails in release mode, so build a debug version
-    macos:
-      xcode: "10.0.0"
-    environment:
-      - HERMES_WS_DIR: /tmp/hermes
-      - TERM: dumb
-      # Homebrew currently breaks while updating:
-      # https://discuss.circleci.com/t/brew-install-fails-while-updating/32992
-      - HOMEBREW_NO_AUTO_UPDATE: 1
+    <<: *apple_defaults
```
Comment (rolsen949) [inline .circleci/config.yml:314]:
```suggestion
    <<: *apple_defaults
```

File: CMakeLists.txt [hunk 6]
```diff
@@ -146,6 +146,9 @@ set(HERMES_ENABLE_THREAD_SANITIZER OFF CACHE BOOL
 set(HERMES_ENABLE_FUZZING OFF CACHE BOOL
   "Enable fuzzing")
 
+set(HERMES_ENABLE_TOOLS ON CACHE BOOL
```
Comment (rolsen949) [inline CMakeLists.txt:149]:
```suggestion
set(HERMES_ENABLE_TOOLS ON CACHE BOOL
```
```suggestion
set(HERMES_ENABLE_TOOLS ON CACHE BOOL
```

File: CMakeLists.txt [hunk 7]
```diff
@@ -146,6 +146,9 @@ set(HERMES_ENABLE_THREAD_SANITIZER OFF CACHE BOOL
...
+  "Enable CLI tools")
```
Comment (rolsen949) [inline CMakeLists.txt:150]:
```suggestion
  "Enable CLI tools")
```

Comment (rolsen949) [inline CMakeLists.txt:150]:
```suggestion
  "Enable CLI tools")
```

File: CMakeLists.txt [hunk 8]
```diff
@@ -146,6 +146,9 @@ set(HERMES_ENABLE_THREAD_SANITIZER OFF CACHE BOOL
...
+
```
Comment (rolsen949) [inline CMakeLists.txt:151]:
```suggestion

```

File: CMakeLists.txt [hunk 9]
```diff
@@ -595,14 +598,17 @@ set(CMAKE_CXX_EXTENSIONS OFF)
...
 add_subdirectory(utils/hermes-lit)
-add_subdirectory(tools)
 add_subdirectory(include)
 add_subdirectory(lib)
 add_subdirectory(public)
 add_subdirectory(external)
 add_subdirectory(API)
 add_subdirectory(android/intltest/java/com/facebook/hermes/test)
 
+if(HERMES_ENABLE_TOOLS)
```
Comment (rolsen949) [inline CMakeLists.txt:608]:
```suggestion
if(HERMES_ENABLE_TOOLS)
```

File: CMakeLists.txt [hunk 10]
```diff
@@ -672,41 +682,44 @@ endif()
...
 
-# If the github release should include extra files (like dlls)
-if (HERMES_GITHUB_RESOURCE_DIR STREQUAL "")
-  set(HERMES_GITHUB_EXTRAS "")
-else()
-  if (IS_DIRECTORY ${HERMES_GITHUB_RESOURCE_DIR})
-    file(GLOB HERMES_GITHUB_EXTRAS "${HERMES_GITHUB_RESOURCE_DIR}/*")
+if(HERMES_ENABLE_TOOLS)
+  set(HERMES_CLI_GITHUB_FILE hermes-cli-${HERMES_GITHUB_SYSTEM_NAME}-v${HERMES_RELEASE_VERSION}.tar.gz)
+  set(HERMES_GITHUB_BUNDLE_DIR ${HERMES_BINARY_DIR}/bundle)
```
Comment (rolsen949) [inline CMakeLists.txt:688]:
```suggestion
  set(HERMES_GITHUB_BUNDLE_DIR ${HERMES_BINARY_DIR}/bundle)
```

File: CMakeLists.txt [hunk 11]
```diff
@@ -672,41 +682,44 @@ endif()
...
+
```
Comment (rolsen949) [inline CMakeLists.txt:689]:
```suggestion

```

**Review by rolsen949 — CHANGES_REQUESTED** [review]

**Comment by alloy** [comment]
@rolsen949 What is this all about? (Are you a bot?)

**Comment by alloy** [comment]
@Huxpro This needs re-importing btw.
